### PR TITLE
YUPANA-86 - making use of iterator while deleting outdated records

### PR DESCRIPTION
**Pull request by kgaikwad**
i.e reportarchive & reportslicearchive records

### CODE REVIEW

**Review by beav — APPROVED** [review]
I have not tested this but the logic looks good to me.

File: yupana/config/settings/base.py [hunk 1]
```diff
@@ -78,6 +78,12 @@
 GARBAGE_COLLECTION_INTERVAL = ENVIRONMENT.get_value(
     'GARBAGE_COLLECTION_INTERVAL', default=604800)
 
+# this is the chunk size used while iterating reports records while GC job run
+# the default is set to 2000
+CHUNK_SIZE_FOR_REPORTS = ENVIRONMENT.get_value(
```
Comment (kgaikwad) [inline yupana/config/settings/base.py:83]: @beav - 
https://docs.djangoproject.com/en/2.2/ref/models/querysets/#iterator- default value is `2000` 
Here it mentioned that 
`Assuming rows of 10-20 columns with a mix of textual and numeric data, 2000 is going to fetch less than 100KB of data, which seems a good compromise between the number of rows transferred and the data discarded if the loop is exited early.`

Comment (beav) [reply yupana/config/settings/base.py:83]: that works. If 2000 still causes the issue we can lower it easily via configuration.

**Comment by kgaikwad** [comment]
Thank you @beav for review! Merging this change.

**Pull request merged**